### ISSUE TITLE
Add Makefile

### ISSUE THREAD

**Issue by oke-aditya** [oke-aditya/template_python#16]
## 🚀 Feature
This has become quite common practice to have Makefile.

I have no much clue how this works and how it is used. (Except we do `make` and this works)

[Here](https://github.com/nateraw/templates/blob/master/custom_package_dev/Makefile) is a template Makefile I found.

Can somehow we can incorporate this?

**Comment by janaSunrise** [comment]
I can work on this 🙌🏼 
Can I get some more context on what to create the makefile for?

**Comment by oke-aditya** [comment]
I would like to hear proposal what a makefile can be made for in this context.
I have never created one yet, so I have little less idea.

**Comment by janaSunrise** [comment]
Makefile is usually used with C, C++ programs for compiling, and basically works like a bash script, but for all OS, It basically simplifies a certain thing / command.

**Comment by oke-aditya** [comment]
Great, how could it possiblu simplify setup in python project?

**Comment by janaSunrise** [comment]
Not too sure, but we could automate things as mentioned in my PR with the script I'll be making, once everything's resolved. Hence `Makefile` is useless unless you have C extensions in this template

**Comment by oke-aditya** [comment]
Feel free to let me know if this issue is invalid.

**Comment by janaSunrise** [comment]
Yeah, We can close this and stick with the startup script. But I think to make it interactive, having external dependencies is fine? Not sure, Lemme know your opinion
